browser_patches/firefox-beta/juggler/NetworkObserver.js

```diff
@@ -863,19 +863,30 @@ function setPostData(httpChannel, postData, headers) {
   const body = atob(postData);
   synthesized.setData(body, body.length);
 
-  const overriddenHeader = (lowerCaseName, defaultValue) => {
+  const overriddenHeader = (lowerCaseName) => {
     if (headers) {
       for (const header of headers) {
         if (header.name.toLowerCase() === lowerCaseName) {
           return header.value;
         }
       }
     }
-    return defaultValue;
+    return undefined;
   }
   // Clear content-length, so that upload stream resets it.
   httpChannel.setRequestHeader('content-length', '', false /* merge */);
-  httpChannel.explicitSetUploadStream(synthesized, overriddenHeader('content-type', 'application/octet-stream'), -1, httpChannel.requestMethod, false);
+  let contentType = overriddenHeader('content-type');
+  if (contentType === undefined) {
+    try {
+      contentType = httpChannel.getRequestHeader('content-encoding');
```

Comment:
Is this correct? What happens with gzipped req/resp? Would we report content-type as `gzip` b/c of this line? That doesn't sound correct.

Reply:
Yes this is the intended behavior unless content-type is explicitly overridden we use the one from the original request. That's what we do in other browsers too.

Reply:
Do you suggest something different?

Reply:
Oh, you meant the typo 'content-encoding' instead of 'content-type', fixed.